### PR TITLE
[DASH-2397] Add url validation for CNAME field, display URL when Host…

### PR DESCRIPTION
**Pull request by Flixow**
…ing name is valid domain

### CODE REVIEW

**Comment by Flixow** [comment]
image

image

**Comment by Flixow** [comment]
image

File: src/apps/Hosting/HostingDialog.js [hunk 1]
```diff
@@ -19,23 +19,39 @@ const CreateHostingDialog = React.createClass({
...
+        length: {
+          maximum: 256
+        }
       }
+    };
+
+    if (!isCnameValid) {
+      validateObj.cname = {
+        inclusion: {
+          message: '^CNAME is not a valid url '
```
Comment (wpasinski) [inline src/apps/Hosting/HostingDialog.js]: _CNAME is not a valid domain_?

**Comment by gh3r** [comment]
Apply comments and LGTM

File: src/apps/Hosting/HostingDialog.js (hunk 1 above)
Comment (Flixow) [inline src/apps/Hosting/HostingDialog.js]: I think that `url` is more clearly. Sb else?

Comment (tomsonpl) [inline src/apps/Hosting/HostingDialog.js]: I think that CNAME has kind of the same meaning as URL / ALIAS.  So probably domain should go here, but I am not an expert (yet ;p)

File: src/apps/Hosting/HostingStore.js [hunk 2]
```diff
@@ -55,10 +55,12 @@ export default Reflux.createStore({
   },
 
   onCreateHostingCompleted(payload) {
+    this.refreshData();
```
Comment (patrykkopycinski) [inline src/apps/Hosting/HostingStore.js]: `refreshData` should be always at the of the method

File: src/apps/Hosting/HostingListItemLinks.js [hunk 3]
```diff
@@ -3,31 +3,35 @@ import { withRouter } from 'react-router';
...
     }
 
-    return domain;
+    return `https://${domain}--${params.instanceName}.syncano.site`;
   };
 
   return (
     <div>
-      {_.map(items, (item) => (
+      {_.map(items, (item, index) => (
```
Comment (patrykkopycinski) [inline src/apps/Hosting/HostingListItemLinks.js]: don't use `index` as an item key

File: src/apps/Hosting/HostingListItemLinks.js [hunk 4]
```diff
@@ -3,31 +3,35 @@ import { withRouter } from 'react-router';
...
 import { LinkWithIcon } from '../../common';
 
-const HostingListItemLinks = ({ hostingLabel, items, isDefault, params }) => {
+const HostingListItemLinks = ({ items, isDefault, params }) => {
   const styles = {
     item: {
       padding: '4px 0'
     }
   };
+  let isCnameFounded = false;
```
Comment (patrykkopycinski) [inline src/apps/Hosting/HostingListItemLinks.js]: what it does mean?

**Comment by gh3r** [comment]
Checked once more after fixes, LGTM 👍

**Pull request merged**